jdtls.ext/com.microsoft.jdtls.ext.core/src/org/eclipse/jdt/internal/jarpackager/JarPackageUtil.java

```diff
@@ -0,0 +1,356 @@
+package org.eclipse.jdt.internal.jarpackager;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Set;
+import java.util.jar.JarEntry;
+import java.util.jar.JarOutputStream;
+import java.util.zip.CRC32;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipException;
+import java.util.zip.ZipFile;
+
+import org.eclipse.core.runtime.*;
+import org.eclipse.jdt.internal.core.manipulation.JavaManipulationPlugin;
+import org.eclipse.jdt.internal.ui.IJavaStatusConstants;
+
+public class JarPackageUtil {
```

Comment:
From my look, writeArchive and writeFile will be the main interfaces exposed for the outer. Is it necessary to keep other methods being public?